articles/151_node_to_participant_mapping.md

```diff
@@ -0,0 +1,241 @@
+Furthermore, the maximum number of `Domain participants` is rather small.
+For example, [RTI connext](https://community.rti.com/kb/what-maximum-number-participants-domain) is limited to 120 participants per domain.
+
+## Proposal
+
+The goal of this proposal is to improve overall performance by avoiding the creation of one `Domain Participant` per `Node`.
+API changes will be avoided, if possible.
+
+### What is a participant mapped to?
+
+There are two main alternatives:
+- One participant per process.
+- One participant per context.
+
+The second approach allows more flexibility.
+Considering that by default there's only one context per process, it wouldn't affect the case where each node runs in its own process.
+And even in the case that multiple nodes are running in a single process it allows to group them into different context - ranging from a separate context for each node, over grouping a few nodes in the same context, to using a single context for all nodes.
+
+### What is a Node now?
+
+There's no lightweight DDS equivalent of a ROS `Node`, so these must be implemented on top of it.
+A `Node` should be able to:
+- Create other entities as `Publishers`, `Subscriptions`, `Services` and `Clients`.
+  `Nodes` should own those entities, that is to say, those entity shouldn't outlive a `Node`.
+- List all its entities.
+
+For all the entities, it should be possible to get the `Node` that created them.
+Each `Participant` should store all the information needed about its nodes, and communicate it other `Participants`.
+
+### How `Node` information is communicated?
+
+#### Using a topic
+
+The name of all the available `Nodes`, and its `Publishers`, `Subscriptions`, `Services`, `Clients` should be available for every `Participant`.
+This information can be communicated using a `topic`.
+That topic will be an implementation detail and hidden to the user (i.e.: the `rt/` prefix won't be added to this `DDS topic`).
+
+A message could be sent for:
+- Each `Node`
+- Each `Participant`
+
+The second option reduces the amount of messages.
+It also allow organizing the data using the `Participant` GUID as the key.
+It's not possible to organize the data using the `Node` name as a key, because it can collide.
+`Node` name uniqueness can be enforced using a collision resolution mechanism, but it can be detected beforehand (i.e.: this information will be needed by the resolution mechanism).
+In the following, the second option will be considered.
+
+##### State Message
+
+Each `Participant` will send a message representing their state.
+A keyed topic could be used for communicating it.
+The `Participant` GUID can be used as the key.
+This helps for keeping only one message per `Participant` in the history (see [QoS for communicating node information](#QoS-for-communicating-node-information)).
+The rest of the message will be a sequence of with the information of each node.
+That message should contain the `Node` name, and four sequences:
```

Comment:
It's not clear to me if this is a message per Node, or a message with a list of nodes (I think it's the latter):
```suggestion
For each `Node`, the message should contain the `Node` name, and four sequences:
```